### PR TITLE
Add tests for IEMOP production

### PR DESCRIPTION
**Pull request by pierresegonne**
## Issue

`IEMOP` parser was not tested

## Description

* Adds test for production for all 3 zones (snapshot based on mocks)

Could one of you do the same for exchanges please??

### Double check

- [x] I have tested my parser changes locally with `poetry run test_parser "zone_key"`
- [x] I have run `pnpx prettier --write .` and `poetry run format` to format my changes.

### CODE REVIEW

File: parsers/IEMOP.py [hunk 1]
```diff
@@ -477,10 +480,14 @@ def get_all_market_reports_items(
         "page": "1",
         "post_id": KIND_TO_POST_ID[kind],
     }
-    r = requests.post(
-        "https://www.iemop.ph/wp-admin/admin-ajax.php", data=form_data, verify=False
-    )
-    id_to_items = eval(r.text)["data"]
```
Comment (pierresegonne) [inline parsers/IEMOP.py:483]: I don't know how this got accepted before - but using `eval` is a very bad practice, especially if it's just `json` and we can run `json.loads`

Comment (VIKTORVAV99) [reply parsers/IEMOP.py:483]: If it's actually JSON we can just do r.json() and then grab the data dict/list.

**Comment by pierresegonne** [comment]
Guys, quick ping, the PR should be pretty quick to review (the additions are snapshots)

**Comment by VIKTORVAV99** [comment]
> Guys, quick ping, the PR should be pretty quick to review (the additions are snapshots)

Will review before lunch (and get some exchange snapshots in if I got the time).

**Review by mathilde-daugy — APPROVED** [review]

**Review by unitrium — APPROVED** [review]
Looks good! One small suggestion about the test.

File: parsers/IEMOP.py [hunk 2]
```diff
@@ -477,10 +479,14 @@ def get_all_market_reports_items(
...
+    r = requests.post(REPORTS_ADMIN_URL, data=form_data, verify=False)
+    id_to_items = r.json().get("data", {})
+    if not id_to_items:
+        raise ParserException(
+            parser="PH.py",
```
Comment (VIKTORVAV99) [inline parsers/IEMOP.py:486]:
```suggestion
            parser="IEMOP.py",
```

**Review by VIKTORVAV99 — APPROVED** [review]
Looks good but I'd like to make one small change to the exemption logging.

(Don't have to time to add the snapshots right now unfortunately)

**Pull request merged**